### ISSUE TITLE
Keep Java code line number in exception stack trace element related to renderMap method

### ISSUE THREAD

**Issue by izogfif** [casid/jte#95]
TL;DR: in jte-runtime inside `gg.jte.runtime.TemplateClassLoader.rewriteStackTrace` method check if exception happened inside `renderMap` method. If so, do not rewrite this stack element.

Better yet, map to the proper `@param TypeName paramName` line of .jte file.

**Rationalization**

I'm in the process of migration from JSP .tag-files to .jte ones. I'm using JTE bridge tag in existing .jsp files. Sometimes JSP evaluates `${a * b}` where both `a` and `b` of type `java.lang.Integer` to result of type `java.lang.Long`. And this value is passed inside .jte tag that expects value to be of type `java.lang.Integer` (`@param int paramName`). Since the type cast happens inside `renderMap` method of the class generated by JTE, all lines of `renderMap` method are "mapped" to the last line of .jte template. In server logs I see something like
<pre>
Caused by: java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at gg.jte.generated.ondemand.tag.JtesampleTagGenerated.renderMap(tag/sampleTag.jte:208) ~[?:?] 
</pre>

And since line 208 in `sampleTag.jte` is empty, I am left wondering what's going on and why is there a type cast in that line in the first place.

My guess was that maybe there is an issue in parameter type. There are many `int` parameters accepted by the tag, and to find out which one is the culprit, I had to attach remote debugger and put the breakpoint inside `gg.jte.runtime.Template.renderMap` where I can read the original (not rewritten) stack trace. And sure enough, it was inside `renderMap` line
<pre>int position = (int)params.get("position");</pre>
which, obviously, was generated from
<pre>@param int position</pre>
and that line is actually line 13 in `sampleTag.jte` file. I expected to see 
<pre>at gg.jte.generated.ondemand.tag.JtesampleTagGenerated.renderMap(tag/sampleTag.jte:13) ~[?:?] </pre>
instead. Or at least 
<pre>gg.jte.generated.ondemand.tag.JtesampleTagGenerated.renderMap(JtesampleTagGenerated.java:306)</pre>
which would be less convenient, but still more helpful than "a class cast exception happened at an empty line of your .jte template" that I'm currently getting in server error logs.

**Comment by casid** [comment]
@izogfif thanks for reporting this and the excellent suggestion! And I'm sorry for the debugging adventures you've had with this!

I've enhanced the code generation to add debug template line information for the renderMap method as you suggested. I added a unit test for your case and it now provides the template line number of the `@param` where the class cast exception happens 👍

**Comment by izogfif** [comment]
@casid I've just checked. It works (points to the `@param` line in .jte file). Thanks.
